File: fixtures/scale/dart/large/lib/ciphers/bifid/bifid_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'bifid_cipher.dart';
import 'bifid_key.dart';
import 'bifid_vectors.dart';

class BifidSuite implements CipherSuite {
  @override
  String name() => 'bifid';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = BifidCipher(BifidKey.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in bifidSamples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/bifid/BifidSuite.scala
package cryptobench.ciphers.bifid

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.RoundTrip

final class BifidSuite extends CipherSuite {
  override def name: String = "bifid"

  override def run(): SuiteResult = {
    val cipher: Cipher = new BifidCipher(BifidKey.default())
    var passed = 0
    var failed = 0
    val start = System.nanoTime()
    for (sample <- BifidVectors.samples()) {
      if (RoundTrip.check(cipher, sample)) passed += 1 else failed += 1
    }
    SuiteResult(name, passed, failed, System.nanoTime() - start)
  }
}
